projects/saturn/src/main/java/io/fairspace/saturn/services/metadata/MetadataService.java

```diff
@@ -264,8 +263,11 @@ private void logUpdates(Set<Resource> updatedResources) {
     private void validate(Model before, Model after, Model modelToRemove, Model modelToAdd) {
         modelToAdd.listSubjects()
                 .andThen(modelToRemove.listSubjects())
-                .filterDrop(s -> permissions.canWriteMetadata(s.inModel(before)))
-                .forEachRemaining(s -> {
+                .filterDrop(s -> {
+                    var type = getType(s.inModel(before)) != null ? getType(s.inModel(before)) : getType(s.inModel(after));
```

Comment:
```suggestion
                    var type = getType(s.inModel(before)) != null ?? getType(s.inModel(after));
```

Reply:
getType(s.inModel(after))

I don't know if this will ever occur, since s comes from 'modelToRemove'

Reply:
` var type = getType(s.inModel(before)) != null ?? getType(s.inModel(after));`
unfortunately you can't do this in Java